baseband/vdif/base.py

```diff
@@ -444,41 +442,46 @@ class VDIFStreamWriter(VDIFStreamBase, VLBIStreamWriterBase, VDIFFileWriter):
     time : `~astropy.time.Time`
         As an alternative, one can pass on ``ref_epoch`` and ``seconds``.
     nchan : int, optional
-        Number of FFT channels within stream (default 1).
+        Number of channels within stream (default 1).
         Note: that different # of channels per thread is not supported.
     complex_data : bool
-        Whether data is complex
+        Whether data is complex.
     bps : int
         Bits per sample (or real, imaginary component).
     samples_per_frame : int
-        Number of complete samples in a given frame.  As an alternative, use
-        ``frame_length``, the number  of long words for header plus payload.
-        For some edv, this number is fixed (e.g., ``frame_length=629`` for
-        edv=3, which corresponds to 20000 real 2-bit samples per frame).
+        Number of complete samples per frame.  As an alternative, use
+        ``frame_length``, the number of long words for header plus payload.
+        For some EDV, this number is fixed (e.g., ``frame_length=629`` for
+        ``edv=3``, which corresponds to 20000 real 2-bit samples per frame).
     station : 2 characters
         Or unsigned 2-byte integer.
     edv : {`False`, 0, 1, 2, 3, 4, 0xab}
         Extended Data Version.
-    bandwidth : `~astropy.units.Quantity`
-        In frequency units.  Sufficient for `edv` 1, 3, or 4 to determine the
-        frames per second.
+    framerate : `~astropy.units.Quantity`, optional
+        In frequency units.  Available for EDV 1 and 3.
+    bandwidth : `~astropy.units.Quantity`, optional
+        In frequency units.  Available for EDV 1 and 3, and sufficient
+        to determine the frame rate.
     """
-    def __init__(self, raw, nthread=1, frames_per_second=None,
-                 sample_rate=None, header=None, squeeze=True, **kwargs):
+    def __init__(self, raw, nthread=1, sample_rate=None, header=None,
+                 squeeze=True, **kwargs):
         if header is None:
             header = VDIFHeader.fromvalues(**kwargs)
         # No frame sets yet exist, so generate a sample shape from values.
         super(VDIFStreamWriter, self).__init__(
-            raw, header, range(nthread), frames_per_second=frames_per_second,
-            sample_rate=sample_rate, squeeze=squeeze)
+            raw, header, range(nthread), sample_rate=sample_rate,
+            squeeze=squeeze)
         # Set framerate and thus bandwidth in the header, if not set already.
         try:
```

Comment:
Another thing that will become nicer with `header.samplerate`

Reply:
I appended an assert line to the try/except/else block to ensure that the header's `sample_rate` is same as the stream reader's (it's possible to pass a complete `header0`, or keywords `sampling_rate` and `sampling unit`, while separately passing a `sample_shape`).